pkg/scheduler/api/pod_status/pod_status.go

```diff
@@ -44,10 +45,21 @@ const (
 	Deleted
 )
 
+// Status group constants using bitwise OR
+const (
+	activeUsedStatuses      = Allocated | Pipelined | Binding | Bound | Running | Releasing
```

Comment:
How is it working? Allocated is not including Pending and Gated... (3 => 11 in binary, which is Pending (01) and Gated (10) 
Can you add unit tests to all the functions that you changed here? only because it is not a trivial refactor.

Reply:
The | operator is bitwise OR, and each status is a power of two. So, say Allocated = 001, Pipelined = 010, Binding = 100, the OR between them is 111.
Added a test for `TestIsAliveStatus`, should be enough to validate this functionality